Tensile/SolutionStructs.py

```diff
@@ -2639,6 +2639,12 @@ def assignDerivedParameters(state):
     if state["WavefrontSize"] == 32 and state["KernelLanguage"] == "Source":
       reject(state, "WavefrontSize=32 not yet supported for source kernels.")
 
+    if state["KernelLanguage"] == "Source" \
+        and state["ProblemType"]["DataType"].isHalf() \
+        and state["LoopDoWhile"] \
+        and not state["AssertSizeEqual"]:
```

Comment:
Should we specify all dimensions?
I feel it is not enough to just check if AssertSizeEqual is defined or not.
We can specify keys of 0,1,2,3 corresponding to each dimension.

Reply:
This particular bug was on dimension 0 - I could add a check specifically for that one if you think we need it, my main goal was to make sure the tuner was checking limitations and validating if using these parameters. I don't fully know the limitations of the LoopDoWhile option, so I don't yet know what to add beyond that. I've made another ticket to investigate LoopDoWhile errors, and potentially generalize if we want to support the feature.

Reply:
> This particular bug was on dimension 0 - I could add a check specifically for that one if you think we need it, my main goal was to make sure the tuner was checking limitations and validating if using these parameters. I don't fully know the limitations of the LoopDoWhile option, so I don't yet know what to add beyond that. I've made another ticket to investigate LoopDoWhile errors, and potentially generalize if we want to support the feature.

I prefer to specify dimension 0 if we already know the bug is related only to dim 0.

Reply:
Updated